docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+
+- [Connection Scoring](#connection-scoring) - calculated during handshake, affects the peers we connect to
+- [Message Scoring](#message-scoring) - tracks operators behavior w.r.t incoming IBFT messages
+- [Pubsub Scoring](#pubsub-scoring) - calculated by libp2p, tracks over gossip tasks and pubsub related metrics
+
+For more info please refer to the linked sections.
+
+------
+
+
+## Wire
+
+Network interaction includes several types of protocols:
+
+## Consensus Protocol
+
+`IBFT`/`QBFT` consensus protocol is used to govern `SSV` network.
+`IBFT` ensures that consensus can be reached by a committee of `n` 
+operator nodes while tolerating a certain amount of `f` faulty nodes as defined by `n ≥ 3f + 1`.
+
+As part of the algorithm, nodes are exchanging messages with other nodes in the committee. \
+Once the committee reaches consensus, the nodes will publish the decided message across the network.
+
+Consensus messages are being sent in the network over pubsub topics (see [subnets](#subnets))
+
+More information regarding the protocol can be found in [iBFT annotated paper (By Blox)](/ibft/IBFT.md)
+
+### Message Structure
+
+`SignedMessage` is a wrapper for IBFT messages, it holds a message and its signature with a list of signer IDs:
+
+
+<details>
+  <summary><b>protobuf</b></summary>
+  
+  ```protobuf
+  syntax = "proto3";
+  import "gogo.proto";
+  
+  // SignedMessage holds a message and it's corresponding signature
+  message SignedMessage {
+    // message is the IBFT message
+    Message message            = 1 [(gogoproto.nullable) = false];
+    // signature is a signature of the IBFT message
+    bytes signature            = 2 [(gogoproto.nullable) = false];
+    // signer_ids is a sorted list of the IDs of the signing operators
+    repeated uint64 signer_ids = 3;
+  }
+  
+  // Message represents an IBFT message
+  message Message {
+    // type is the IBFT state / stage
+    Stage type        = 1;
+    // round is the current round where the message was sent
+    uint64 round      = 2;
+    // identifier is the message identifier
+    bytes identifier      = 3;
+    // sequence number is an incremental number for each instance, much like a block number would be in a blockchain
+    uint64 seq_number = 4;
+    // value holds the message data in bytes
+    bytes value       = 5;
+  }
+  ```
+</details>
+
+<details>
+  <summary><b>JSON example</b></summary>
+
+  ```json
+  {
+    "message": {
+      "type": 3,
+      "round": 1,
+      "identifier": "OTFiZGZjOWQxYzU4NzZkYTEwY...",
+      "seq_number": 28276,
+      "value": "mB0aAAAAAAA4AAAAAAAAADpTC1djq..."
+    },
+    "signature": "jrB0+Z9zyzzVaUpDMTlCt6Om9mj...",
+    "signer_ids": [2, 3, 4]
+  }
+  ```
+</details>
+
+**NOTE:** 
+- all pubsub messages in the network are wrapped with libp2p's message structure
+- `signer_ids` must be sorted, to allow hashing the entire message
+
+---
+
+## Sync Protocols
```

Comment:
I think we can leave sync messages to:
1) get highest sequence for a specific QBFT instance
2) get historical decided.

We can have a flag in the node for saving/ not saving historical decided messages.
Both exporter and a regular node (operator and not operator) can comply with those api/ rpc calls.

Should we make public what the node supports in the handshake?

Reply:
Why to reduce the last round change?

Handshake won't be needed, libp2p supports this case OOTB with [protocol negotiation](https://github.com/libp2p/specs/blob/master/connections/README.md#protocol-negotiation).

Reply:
Also, regards operators storing decided messages, do you refer to their validators or all the validators in the subnets they are part of?